python/cuml/manifold/umap.pyx

```diff
@@ -348,19 +352,25 @@ class UMAP(Base):
 
         self.umap_params = <size_t> umap_params
 
+        if self.MNMG:
+            self.state = self.__getstate__()
+            free(umap_params)
+            self.umap_params = 0
+
         self.callback = callback  # prevent callback destruction
         self.X_m = None
         self.embedding_ = None
 
     def __getstate__(self):
-        state = self.__dict__.copy()
-
-        del state['handle']
+        state = dict()
+        if not self.MNMG:
+            state = self.__dict__.copy()
+            del state['handle']
 
         cdef size_t params_t = <size_t>self.umap_params
         cdef UMAPParams* umap_params = <UMAPParams*>params_t
 
-        if hasattr(self, "X_m") and self.X_m is not None:
+        if not self.MNMG and hasattr(self, "X_m") and self.X_m is not None:
             # fit has not yet been called
             state['X_m'] = cudf.DataFrame.from_gpu_matrix(self.X_m)
```

Comment:
This is done automatically when using the cuml array, so we can remove the `__getstate__` and `__setstate__` functions altogether once the `umap_params` are handled lazily.